src/sweeps/params.py

```diff
@@ -63,6 +72,26 @@ def __init__(self, name: str, config: dict):
             None if self.type != HyperParameter.CONSTANT else self.config["value"]
         )
 
+    def coerce(self):
```

Comment:
redo this as a context manager

Reply:
nvm. replace this approach by leaving config/type alone and using shared logic for pdf,cdf, sample, etc.